nightfall-optimist/src/services/challenges.mjs

```diff
@@ -185,63 +164,57 @@ export async function createChallenge(block, transactions, err) {
         const { transactionHashIndex: transactionIndex } = err.metadata;
         // Create a challenge
         const uncompressedProof = transactions[transactionIndex].proof;
-        if (transactions[transactionIndex].transactionType === '0') {
-          txDataToSign = await challengeContractInstance.methods
-            .challengeProofVerification(
-              Block.buildSolidityStruct(block),
-              transactions.map(t => Transaction.buildSolidityStruct(t)),
-              transactionIndex,
-              uncompressedProof,
-              salt,
-            )
-            .encodeABI();
-        } else if (transactions[transactionIndex].transactionType === '2') {
-          // Create a specific challenge for a double_transfer
-          const [historicInput1, historicInput2] = await Promise.all(
-            transactions[transactionIndex].historicRootBlockNumberL2.map(async b => {
-              const historicBlock = await getBlockByBlockNumberL2(b);
-              const historicTxs = await getTransactionsByTransactionHashes(block.transactionHashes);
+        const [historicInput1, historicInput2] = await Promise.all(
+          transactions[transactionIndex].historicRootBlockNumberL2.map(async b => {
+            if (b === 0) {
```

Comment:
How do you tell the difference between the blockNumber == 0 because it's a deposit and a transfer that uses the root from blockNumber 0?